File: js/src/Matrix/RotateImage.js



export const Rotate = (list) => {
    const length = list.length;

    for (let i = 0; i < length; i++) {

        for (let j = i; j < length; j++) {
        
            let temp = list[i][j];
        
            list[i][j] = list[j][i];
        
            list[j][i] = temp;
        
        }
    }

    for (let i = 0; i < length; i++) {

        list[i].reverse();
   
    }

    return list;
}

File: python/matrix/rotate_image.py




def rotate_image(board: list[list[int]]) -> list[list[int]]:

    length = len(board)

    for i in range(length):
        for j in range(i + 1, length): 
            board[i][j], board[j][i] = board[j][i], board[i][j]
    

    for i in range(length):
        board[i].reverse()

    return board
